src/trial.jl

```diff
@@ -1,18 +1,39 @@
 function dag_chain(c::Chain, ip)
```

Comment:
@jpsamaroo  does this look kind of alright?

Reply:
I guess so, as long as it gives you the results you expect :laughing:

Reply:
I tried it on resnet and that seems to work fine. Although, one problem is see already is that it wouldn't look into Chains within Chains all that well and try to schedule entire nested chains as one operation when in reality those would be scheduled separately as a DAG. So maybe the correct thing is to repeatedly call `dag_chain` instead of `pullback`?

Also we'd need a way to express that some operations probably are cheap and not worth transferring data for.

Reply:
> So maybe the correct thing is to repeatedly call dag_chain instead of pullback?

That sounds reasonable to me.

> Also we'd need a way to express that some operations probably are cheap and not worth transferring data for.

Yeah, https://github.com/JuliaParallel/Dagger.jl/pull/221 should eventually fix this (but it needs a lot more work still). Work that's determined to be computationally cheap but has large data transfer costs would be more likely to stay on the same node, while when the opposite it true, the work is more likely to move to another node when the current node is under heavy load.